Userland/Libraries/LibJS/Runtime/ArrayBuffer.cpp

```diff
@@ -118,21 +118,39 @@ void copy_data_block_bytes(ByteBuffer& to_block, u64 to_index, ByteBuffer& from_
     // 7. Return unused.
 }
 
-// 25.1.2.1 AllocateArrayBuffer ( constructor, byteLength ), https://tc39.es/ecma262/#sec-allocatearraybuffer
-ThrowCompletionOr<ArrayBuffer*> allocate_array_buffer(VM& vm, FunctionObject& constructor, size_t byte_length)
+// 25.1.2.1 AllocateArrayBuffer ( constructor, byteLength [ , maxByteLength ] ), https://tc39.es/proposal-resizablearraybuffer/#sec-allocatearraybuffer
+ThrowCompletionOr<ArrayBuffer*> allocate_array_buffer(VM& vm, FunctionObject& constructor, size_t byte_length, Optional<size_t> max_byte_length)
 {
-    // 1. Let obj be ? OrdinaryCreateFromConstructor(constructor, "%ArrayBuffer.prototype%", « [[ArrayBufferData]], [[ArrayBufferByteLength]], [[ArrayBufferDetachKey]] »).
+    // 1. Let slots be « [[ArrayBufferData]], [[ArrayBufferByteLength]], [[ArrayBufferDetachKey]] ».
+
+    // 2. If maxByteLength is present and not empty, then
+    // a. If byteLength > maxByteLength, throw a RangeError exception.
+    // b. Append [[ArrayBufferMaxByteLength]] to slots.
+    if (max_byte_length.has_value() && byte_length > max_byte_length.value())
```

Comment:
This could be implemented closer to spec text with:
```
    // 2. If maxByteLength is present and not empty, then
   if (max_byte_length.has_value()) {
         // a. If byteLength > maxByteLength, throw a RangeError exception
         if (byte_length > max_byte_length.value()))
            return vm.throw_completion<RangeError>(ErrorType::InvalidByteLengthForResizableArrayBuffer, byte_length, max_byte_length.value());
         // b. Append [[ArrayBufferMaxByteLength]] to slots.
     }
```

Probably same with step c.

Reply:
Thanks for the review, i've implemented the reviewed changes :)